app/tutorials/weather-warning/src/main/java/org/eclipse/mosaic/app/tutorial/WeatherServerApp.java

```diff
@@ -31,25 +37,18 @@
  * about certain hazards on the road. The hazard is hard-coded for tutorial purposes,
  * in more realistic scenarios the location would've been updated dynamically.
  */
-public class WeatherServerApp extends AbstractApplication<ServerOperatingSystem> {
+public class WeatherServerApp extends AbstractApplication<ServerOperatingSystem> implements CommunicationApplication {
 
     /**
-     * Send hazard location at this interval, in seconds.
+     * Send warning at this interval, in seconds.
      */
     private final static long INTERVAL = 2 * TIME.SECOND;
 
-    /**
-     * Location of the hazard which causes the route change.
-     */
-    private final static GeoPoint HAZARD_LOCATION = GeoPoint.latLon(52.633047, 13.565314);
 
     /**
-     * Road ID where hazard is located.
+     * Save the last received DEN message for relaying.
      */
-    private final static String HAZARD_ROAD = "311964536_1313885442_2879911873";
-
-    private final static SensorType SENSOR_TYPE = SensorType.ICE;
-    private final static float SPEED = 25 / 3.6f;
+    private Denm lastMessage = null;
```

Comment:
could we rename it to `relayMessage` or something similar more meaningful? "lastMessage" sounds for me more like the last message (in sense of "final", not "most recent").